### ISSUE TITLE
Clang doesn't emit deprecated warnings on deprecated template aliases

### ISSUE THREAD

**Issue by philnik777** [llvm/llvm-project#60181]
Clang warns on deprecated aliases, but not on deprecated template aliases.

```c++
template <class T>
using DeprAlias [[deprecated]] = int;

struct A {
  DeprAlias<int> d; // clang should warn here, but doesn't.
};
```

**Comment by philnik777** [comment]
Looks like this got fixed in Clang 19: https://godbolt.org/z/c6K3eT9en
